packages/test/test-end-to-end-tests/src/test/audience.spec.ts

```diff
@@ -0,0 +1,156 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { Container } from "@fluidframework/container-loader";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { flattenRuntimeOptions } from "./flattenRuntimeOptions";
+
+class TestDataObject extends DataObject {
+    // Maintains a list of clients in the audience.
+    public audienceClientList: Set<string> = new Set();
+
+    public get _root() {
+        return this.root;
+    }
+
+    public get _runtime() {
+        return this.runtime;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+
+    protected async hasInitialized() {
+        this.runtime.getAudience().on("addMember", (clientId: string) => {
+            this.audienceClientList.add(clientId);
+        });
+
+        this.runtime.getAudience().on("removeMember", (clientId: string) => {
+            this.audienceClientList.delete(clientId);
+        });
+
+        const members = this.runtime.getAudience().getMembers();
+        for (const [clientId] of members) {
+            this.audienceClientList.add(clientId);
+        }
+    }
+}
+
+describeNoCompat("Audience correctness", (getTestObjectProvider) => {
```

Comment:
Should be have full compat instead of noCompat?

Reply:
Full compat was breaking because we broke this in the last release. However, with 0.45.2 as N-1, that should work. Let me give it a try.